### PR TITLE
Remove unused resources on Android from brave_resources.pak

### PR DESCRIPTION
**Pull request by AlexeyBarabash**
Resolves https://github.com/brave/brave-browser/issues/30082

These resources were removed as unused on Android and the effect on size reduction is below:

№ | description | apk size | apk size delta | brave_resources.pack size | brave_resources.pack size delta
--- | --- | --- | --- | --- | ---
0 | initial | 158,629,333 |   | 3157651 |  
1 | no IDR_BRAVE_REWARDS_FAVICON | 158,611,657 | -17,676 | 3142559 | -15092
2 | no IDR_BRAVE_COMMON_TOOLBAR_IMG | 158,612,091 | 434 | 3142125 | -434
3 | no IDR_WEBCOMPAT_REPORTER_HTML | 158,612,425 | 334 | 3141791 | -334
4 | no IDR_BRAVE_WELCOME_HTML..PAGE_FAVICON | 158,614,792 | 2,367 | 3139424 | -2367
5 | no brave_welcome_strings.grdp | 158,614,792 | 0 | 3139424 | 0
6 | no IDR_BRAVE_NEW_TAB_HTML--IMG1_IMG4_IMG2 | 158,608,586 | -6,206 | 3137438 | -1986
7 | no IDR_BRAVE_PRIVATE_TAB_IMG..TOR_IMG | 158,604,539 | -4,047 | 3133289 | -4149
8 | no IDR_COMMANDS_HTML | 158,604,983 | 444 | 3132845 | -444
9 | no IDR_PLAYLIST_HTML..PLAYER_HTML | 158,605,681 | 698 | 3132147 | -698
10 | no IDR_BRAVE_BLANK_NEW_TAB_HTML | 158,605,829 | 148 | 3131999 | -148
11 | no IDR_IPFS_HTML | 158,606,121 | 292 | 3131707 | -292
12 | no-IDR_SPEEDREADER_UI_HTML..no-some-strings | 158,579,449 | -26,672 | 3131325 | -382
13 | no IDR_SHIELDS_PANEL_HTML..ICON_64..ICON_64_DISABLED..COOKIE_LIST_OPT_IN_HTML | 158,440,045 | -139,404 | 3123858 | -7467
14 | no IDR_BRAVE_PRIVATE_NEW_TAB_HTML-..many_strings | 158,263,515 | -176,530 | 3123445 | -413
15 | no IDR_BRAVE_WALLET_SOLANA_WEB3_JS_FOR_TEST | 158,163,676 | -99,839 | 3018488 | -104957
16 | no IDR_CHAT_UI_HTML and strings | 158,080,819 | -82,857 | 3018019 | -469
 |  |  |  |  | 
FF | total |   | -548,514 |   | -139632

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
Nothing should be broken.
General test of Android work must be enough.

### CODE REVIEW

**Review by SergeyZhukovsky — APPROVED** [review]
++ for a wallet core side.

**Review by iefremov — APPROVED** [review]
speedreader and chromium_src LGTM

File: browser/sources.gni [hunk 1]
```diff
@@ -323,6 +321,8 @@ if (is_android) {
     "//brave/browser/brave_ads/tooltips",
     "//brave/browser/importer",
     "//brave/browser/ui/bookmark",
+    "//brave/components/ai_chat",
+    "//brave/components/ai_chat:mojom",
```
Comment (nullhook) [inline browser/sources.gni:325]: Including `//brave/components/ai_chat` will have constants.{h|cc} included which has localized strings. Do we want that?

**Review by nullhook — CHANGES_REQUESTED** [review]

File: browser/sources.gni (hunk 1 above)
Comment (nullhook) [reply browser/sources.gni:325]: Makes sense.

**Review by nullhook — APPROVED** [review]
AIChat resources lgtm

**Review by mkarolin — APPROVED** [review]
`grd(p)` ++

**Pull request merged**